### ISSUE TITLE
Let user know that net_write_timeout limits the amount of time you have after crtl-c is pressed before connection close and resuming will not be possible anymore

### ISSUE THREAD

**Issue by utdrmac** [mydumper/mydumper#1523]
**Describe the bug**
During a mydumper backup, ctrl-c was pressed to pause the dump. About 1 minute later, 'N' was pressed to resume the dump. mydumper appeared hung for over 5m before we had to kill mydumper.

**To Reproduce**
```
# mydumper -h srq-ngdb1 -u percona -a --triggers --routines --hex-blob -v 3 --skip-definer --rows 1000:1000:0 --threads 4 --ssl-mode=DISABLED --compress=ZSTD
** Message: 18:40:09.194: MyDumper backup version: 0.16.3-2
Enter MySQL Password:
** Message: 18:40:14.164: Using 4 dumper threads
** Message: 18:40:14.412: Connection via default library settings using password:
        Host: srq-ngdb1
        User: percona
** Message: 18:40:14.991: Connected to MySQL 5.7.15
** Message: 18:40:15.049: Started dump at: 2024-06-04 18:40:15
** Message: 18:40:15.167: Acquiring Global lock
** Message: 18:40:15.338: FLUSH NO_WRITE_TO_BINLOG TABLES: OK
** Message: 18:40:15.397: FLUSH TABLES WITH READ LOCK: OK
** Message: 18:40:15.454: conf created
** Message: 18:40:15.454: End job creation
** Message: 18:40:15.454: Creating workers
** Message: 18:40:15.454: Starting Non-InnoDB tables
** Message: 18:40:15.744: Thread 4: connected using MySQL connection ID 12020991
** Message: 18:40:15.744: Thread 1: connected using MySQL connection ID 12020993
** Message: 18:40:15.744: Thread 3: connected using MySQL connection ID 12020992
** Message: 18:40:15.744: Thread 2: connected using MySQL connection ID 12020994
...
** Message: 20:36:00.872: Thread 4: dumping data for `dedede`.`auditevent`  WHERE (3469923015 <= `ID` AND `ID` <= 3469955014)       into export-20240604-184014/dedede.auditevent.00000.sql| Remaining jobs in this table: 4 All remaining jobs: 27
** Message: 20:36:00.958: Thread 3: dumping data for `dedede`.`auditevent`  WHERE (3500161069 <= `ID` AND `ID` <= 3500193068)       into export-20240604-184014/dedede.auditevent.00002.sql| Remaining jobs in this table: 4 All remaining jobs: 27
** Message: 20:36:01.875: Thread 2: dumping data for `dedede`.`auditevent`  WHERE (3560348203 <= `ID` AND `ID` <= 3560380202)       into export-20240604-184014/dedede.auditevent.00003.sql| Remaining jobs in this table: 4 All remaining jobs: 27
** Message: 20:36:02.664: Thread 1: dumping data for `dedede`.`auditevent`  WHERE (3530414138 <= `ID` AND `ID` <= 3530446137)       into export-20240604-184014/dedede.auditevent.00001.sql| Remaining jobs in this table: 4 All remaining jobs: 27
^C2024-06-04 20:36:02.857370: Ctrl+c detected! Are you sure you want to cancel(Y/N)?N
2024-06-04 20:36:45.543153: Resuming backup

(mydumper hung here. No activity in 'show processlist' in another terminal)
```

**Expected behavior**
It is expected that 'N' answer should resume the backup since we did not want to cancel.

**Environment (please complete the following information):**
 - OS version: Ubuntu 22.04
 - MyDumper version: 0.16.3-2
 - MySQL version: 5.7.15

**Comment by utdrmac** [comment]
400 tables total across 3 databases; total dataset size 60GB.

**Comment by utdrmac** [comment]
Attempting to kill results in another hang

```
(ctrl-z inside mydumper to get to shell)

# ps -Af | grep mydumper
root       37148   35988  4 18:40 pts/2    00:05:46 mydumper -h srq-ngdb1 -u percona -a --triggers --routines --hex-blob -v 3 --skip-definer --rows 1000:1000:0 --threads 4 --ssl-mode=DISABLED --compress=ZSTD
root       38437   35988  0 20:46 pts/2    00:00:00 grep --color=auto mydumper
root@ip-172-30-1-21:/opt/mysql/temp#
root@ip-172-30-1-21:/opt/mysql/temp# kill 37148
root@ip-172-30-1-21:/opt/mysql/temp#
root@ip-172-30-1-21:/opt/mysql/temp# fg
mydumper -h srq-ngdb1 -u percona -a --triggers --routines --hex-blob -v 3 --skip-definer --rows 1000:1000:0 --threads 4 --ssl-mode=DISABLED --compress=ZSTD
** Message: 20:46:52.283: Shutting down gracefully
```

**Comment by davidducos** [comment]
Hi @utdrmac, 
I reviewed your issue and I was able to reproduce it. Honestly, I don't think that we can consider this as a bug, as it will depend on the configuration on the database side. The problem is that the connection is reaching to net_write_timeout and that is why mydumper is hanging silently. We never considered this variable important when we implemented this feature, but now I see the problem and I think that we should at least inform the user that we are not be able to continue if connection is close.
